src/array.rs

```diff
@@ -41,8 +53,30 @@ impl<T: RuntimeType> Array<T> {
     }
 }
 
-impl<T> Drop for Array<T> {
+impl<T: RuntimeType> std::ops::Deref for Array<T> {
+    type Target = [T];
+
+    fn deref(&self) -> &[T] {
+        if self.data.is_null() {
+            return &[];
+        }
+
+        unsafe { std::slice::from_raw_parts(self.data, self.len as usize) }
+    }
+}
+
+impl<T: RuntimeType> std::ops::DerefMut for Array<T> {
+    fn deref_mut(&mut self) -> &mut [T] {
+        if self.data.is_null() {
+            return &mut [];
+        }
+
+        unsafe { std::slice::from_raw_parts_mut(self.data, self.len as usize) }
+    }
+}
+
+impl<T: RuntimeType> Drop for Array<T> {
     fn drop(&mut self) {
-        // TODO: CoTaskMemFree
+        self.clear();
     }
 }
```

Comment:
Can we add some unit tests for Array?

Reply:
I have some basic tests that cover the various issues here in the tests/boxing.rs - its difficult to cover all the edge cases without #213 - that's up next.